### PR TITLE
journal: Trivial cleanup

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/journal/Journaler.cc [hunk 1]
```diff
@@ -113,22 +116,24 @@ Journaler::~Journaler() {
     m_metadata->put();
     m_metadata = nullptr;
   }
-  assert(m_trimmer == nullptr);
-  assert(m_player == nullptr);
-  assert(m_recorder == nullptr);
+  assert(!m_trimmer);
```
Comment (dillaman) [inline src/journal/Journaler.cc]: Nit: I prefer the original way (i.e. don't treat non-booleans as booleans).

File: src/journal/Journaler.cc [hunk 2]
```diff
@@ -231,6 +237,7 @@ void Journaler::create(uint8_t order, uint8_t splay_width,
   int r = m_header_ioctx.aio_operate(m_header_oid, comp, &op);
   assert(r == 0);
   comp->release();
+  comp = nullptr;
```
Comment (dillaman) [inline src/journal/Journaler.cc]: Nit: it's a local variable going out-of-scope immediately

File: src/journal/Journaler.cc [hunk 3]
```diff
@@ -444,10 +453,10 @@ void Journaler::get_metadata(uint8_t *order, uint8_t *splay_width,
 std::ostream &operator<<(std::ostream &os,
 			 const Journaler &journaler) {
   os << "[metadata=";
-  if (journaler.m_metadata != NULL) {
+  if (journaler.m_metadata) {
     os << *journaler.m_metadata;
   } else {
-    os << "NULL";
+    os << "nullptr";
```
Comment (dillaman) [inline src/journal/Journaler.cc]: Nit: find and replace fail?

**Comment by shinobu-x** [comment]
@dillaman Please check, when you get a chance.

File: src/journal/Journaler.cc [hunk 4]
```diff
@@ -444,10 +451,10 @@ void Journaler::get_metadata(uint8_t *order, uint8_t *splay_width,
 std::ostream &operator<<(std::ostream &os,
 			 const Journaler &journaler) {
   os << "[metadata=";
-  if (journaler.m_metadata != NULL) {
+  if (journaler.m_metadata) {
     os << *journaler.m_metadata;
   } else {
-    os << "NULL";
+    os << "failed to find and replace";
```
Comment (dillaman) [inline src/journal/Journaler.cc]: Nit: in my original comment, I was attempting to say no need to change this line

Comment (shinobu-x) [reply src/journal/Journaler.cc]: @dillaman We should say something here which is more helpful for users, shouldn't we?

Comment (dillaman) [reply src/journal/Journaler.cc]: Sure -- you can say "uninitialized"

**Comment by shinobu-x** [comment]
@dillaman finally i noticed you're absolutely right. i deleted such a stupid change.

**Pull request merged**